data-service/src/test/java/uk/gov/gchq/palisade/service/data/request/DataRequestTest.java

```diff
@@ -0,0 +1,99 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.data.request;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.json.JsonTest;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.boot.test.json.JsonContent;
+import org.springframework.boot.test.json.ObjectContent;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.User;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.resource.impl.FileResource;
+import uk.gov.gchq.palisade.resource.impl.SystemResource;
+import uk.gov.gchq.palisade.rule.Rules;
+import uk.gov.gchq.palisade.service.SimpleConnectionDetail;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@JsonTest
+public class DataRequestTest {
+
+    @Autowired
+    private JacksonTester<DataRequest> jacksonTester;
+
+    /**
+     * Create the object using the builder and then serialise it to a Json string. Test the content of the Json string
+     *
+     * @throws IOException if it fails to parse the object
+     */
+    @Test
+    public void testSerialiseResourceResponseToJson() throws IOException {
+        Context context = new Context().purpose("testContext");
+        User user = new User().userId("testUserId");
+        LeafResource resource = new FileResource().id("/test/file.format")
+                .type("java.lang.String")
+                .serialisedFormat("format")
+                .connectionDetail(new SimpleConnectionDetail().serviceName("test-service"))
+                .parent(new SystemResource().id("/test"));
+        Rules rules = new Rules<>();
+
+        DataRequest dataRequest = DataRequest.Builder.create()
+                .withUserId("originalUserID")
+                .withResourceId("originalResourceID")
+                .withContext(context)
+                .withUser(user)
+                .withResource(resource)
+                .withRules(rules);
+
+        JsonContent<DataRequest> dataRequestJsonContent = jacksonTester.write(dataRequest);
+
+        assertThat(dataRequestJsonContent).extractingJsonPathStringValue("$.user.userId.id").isEqualTo("testUserId");
+        assertThat(dataRequestJsonContent).extractingJsonPathStringValue("$.context.contents.purpose").isEqualTo("testContext");
```

Comment:
do we not want to check that the resource details are correctly serialised and that the rules are also correctly serialised?